Providers/Profile.swift

```diff
@@ -722,24 +723,41 @@ public class BrowserProfile: Profile {
             notifySyncing(NotificationProfileDidStartSyncing)
         }
 
-        private func endSyncing(result: Maybe<EngineResults>?) {
+        private func endSyncing(result: SyncOperationResult) {
             // loop through status's and fill sync state
             syncLock.lock()
             defer { syncLock.unlock() }
             log.info("Ending all queued syncs.")
 
-            if let syncResult = result {
-                syncDisplayState = SyncStatusResolver(engineResults: syncResult).resolveResults()
-            } else {
-                syncDisplayState = .Good
+            syncDisplayState = SyncStatusResolver(engineResults: result.engineResults).resolveResults()
+            if AppConstants.MOZ_ADHOC_SYNC_REPORTING {
+                reportAdHocEndSyncingStatus(syncDisplayState, engineResults: result.engineResults)
             }
 
-            reportEndSyncingStatus(syncDisplayState, engineResults: result)
+            reportSyncPingForResult(result)
             notifySyncing(NotificationProfileDidFinishSyncing)
             syncReducer = nil
         }
 
-        private func reportEndSyncingStatus(displayState: SyncDisplayState?, engineResults: Maybe<EngineResults>?) {
+        private func reportSyncPingForResult(opResult: SyncOperationResult) {
+            guard profile.prefs.boolForKey("settings.sendUsageData") ?? true else {
```

Comment:
If this code exists in more than one place in the code, lift it into a method on `profile`?